crates/ruff_python_parser/src/error.rs

```diff
@@ -813,6 +813,41 @@ pub enum UnsupportedSyntaxErrorKind {
     /// [PEG parser rewrite]: https://peps.python.org/pep-0617/
     /// [Python 3.11 release]: https://docs.python.org/3/whatsnew/3.11.html#other-language-changes
     UnparenthesizedUnpackInFor,
+    /// Represents the use of multiple exception names in an except clause without an `as` binding, before Python 3.14.
+    ///
+    /// ## Examples
+    /// Before Python 3.14, catching multiple exceptions required
+    /// parentheses like so:
+    ///
+    /// ```python
+    /// try:
+    ///     ...
+    /// except (ExceptionA, ExceptionB, ExceptionC):
+    ///     ...
+    /// ```
+    ///
+    /// Starting with Python 3.14, thanks to [PEP 758], it was permitted
+    /// to omit the parentheses:
+    ///
+    /// ```python
+    /// try:
+    ///     ...
+    /// except ExceptionA, ExceptionB, ExceptionC:
+    ///     ...
+    /// ```
+    ///
+    /// However, parentheses are still required in the presence of an `as`:
+    ///
+    /// ```python
+    /// try:
+    ///     ...
+    /// except (ExceptionA, ExceptionB, ExceptionC) as e:
+    ///     ...
+    /// ```
+    ///
+    ///
+    /// [PEP 758]: https://peps.python.org/pep-0758/
+    UnparenthesizedExceptExpression,
```

Comment:
bikeshedding: what about `UnparenthesizedExceptionTypes` which is what we also use in the diagnostic message? No strong preference, what you have works as well.